cli/src/torii/routing.rs

```diff
@@ -122,17 +122,19 @@ pub(crate) async fn handle_queries<W: WorldTrait>(
 ) -> Result<Scale<VersionedPaginatedQueryResult>> {
     let (valid_request, filter) = request.validate(&wsv, &query_validator)?;
     let original_result = valid_request.execute(&wsv)?;
-    let total: u64 = original_result
-        .len()
-        .try_into()
-        .map_err(|e: TryFromIntError| QueryError::Conversion(e.to_string()))?;
-
     let result = filter.filter(original_result);
-    let result = QueryResult(if let Value::Vec(value) = result {
-        Value::Vec(value.into_iter().paginate(pagination).collect())
+    let (total, result) = if let Value::Vec(value) = result {
+        (
+            value.len(),
+            Value::Vec(value.into_iter().paginate(pagination).collect()),
+        )
     } else {
```

Comment:
Practical enough, though not enough to deal with the case of nested `Value::Vec`

Reply:
We don't need to deal with the nested structures, see this [comment](https://github.com/hyperledger/iroha/issues/2202#issuecomment-1149995883).